dataservice/api/common/validation.py

```diff
@@ -47,3 +47,54 @@ def enum_validation_generator(_enum, common=True):
         ', '.join([str(el) for el in extended_enum]))
 
     return OneOf(extended_enum, error=error_message)
+
+
+def list_validation_generator(valid_items, items_name='items'):
+    """
+    Return a list validation function
+
+    :param valid_items: valid collection of items to use in validation
+    :type valid_items: list
+    :items_name: the plural identifier of an item in the list. Used in the
+    ValidationError message
+
+    :returns: the validate_list method
+    """
+    def validate_list(input_items):
+        """
+        Check whether all items in the input list exist in another list
+        representing the collection of valid values.
+
+        All list items are converted to strings before
+        doing comparison and validation.
+
+        ** NOTE **
+        This method is used by marshmallow_sqlalchemy.field_for during
+        HTTP request validation. If the HTTP request method is GET,
+        `input_items` will be a list with one element which will be a delimited
+        string representing a list of strings
+        (e.g. ['duo:0000005,duo:0000001']). This is because the value comes
+        from the URL query parameter string. No need to validate this case.
+
+        :param input_items: input collection of items to validate
+        :type input_items: list or str
+        :raises: ValidationError when validation fails
+        """
+        assert isinstance(input_items, list), (
+            'Parameter `input_items` must be a list'
+        )
+        # Don't do validation for this case, just return. See NOTE in docstring
+        if len(input_items) == 1 and any([d in input_items[0]
+                                          for d in [',', ' ', ';']]):
+            return
+
```

Comment:
I am also down for the case insensitive validation. We may be able to lowercase query params and enumeration constants and validate if they match. Once matched, we can map the source value to the controlled value in  enumeration.

Reply:
So I thought about it some more and I've decided against case insensitive validation for a couple reasons: 

- I want to follow the same validation pattern as with enum validation where
we do not do case insensitive validation

- Allowing case insensitive validation in general would cause us to have to change/add 
other code that I don't think is worth it right now. 

Basically, if the validators ignore case then people can put in things like  DNA or dna or DnA. That's fine but then if we want values to "look uniform" in  the db and/or eventually the portal then we have to do something to modify (e.g. title case everything) text values before they get into the db. 

I'm sure there's various ways of doing this (post deserialization in marshmallow, in the base model.py class, etc) but like I said I don't think its worth it to figure this out now.